src/Chronicle/Managers/SagaCoordinator.cs

```diff
@@ -53,7 +53,7 @@ internal sealed class SagaCoordinator : ISagaCoordinator
             {
                 var (isInitialized, state) = await _initializer.TryInitializeAsync(saga, id, message);
 
-                if (!isInitialized)
+                if (!isInitialized || state.State is SagaStates.Completed)
```

Comment:
Just a note, this is useful to avoid redundant processing. Some sagas may require interaction from a user outside of the normal flow - like clicking on a link in an e-mail, which they could do multiple times. I couldn't think of a case where one would want to process a message for a saga after it has already been marked Completed.